smart-contracts/contracts/CosmosBridge.sol

```diff
@@ -242,8 +267,20 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
         );
         lastNonceSubmitted = claimData.nonce;
 
-        // if we are double pegging, then we are going to need to get the address on this chain
-        address tokenAddress = claimData.doublePeg ? sourceAddressToDestinationAddress[claimData.tokenAddress] : claimData.tokenAddress;
+        // if we are double pegging AND we don't control the token, we deploy a new smart contract
+        address tokenAddress;
+        if(claimData.doublePeg && !_isManagedToken(claimData.tokenAddress)) {
+            tokenAddress = _createNewBridgeToken(
+                claimData.tokenSymbol,
+                claimData.tokenName,
+                claimData.tokenAddress,
+                claimData.tokenDecimals,
+                claimData.networkDescriptor
+            );
+        } else {
```

Comment:
We could just make this an else if clause and make sure that this is a double peg tx, then we wouldn't need to do the ternary operator for assigning the tokenAddress.